Gems/ScriptCanvas/Code/Include/ScriptCanvas/Translation/GraphToLuaUtility.cpp

```diff
@@ -23,6 +25,23 @@
 
 namespace GraphToLuaUtilityCpp
 {
+    class ScopedLocale
+    {
+    public:
+        ScopedLocale()
+        {
+            m_previousLocale = std::setlocale(LC_NUMERIC, "en_US.UTF-8");
```

Comment:
I tried doing this with a custom facet and std::locale::global and it it's do_decimal_point() method just refused to get called. This method is the next least invasive way.

Reply:
https://github.com/fengari-lua/fengari/issues/136 FWIW, we're not the only ones dealing with the fact that Lua doesn't necessarily respect locales.